core/src/main/kotlin/Samples/KotlinWebsiteSampleProcessingService.kt

```diff
@@ -0,0 +1,81 @@
+package Samples
+
+import com.google.inject.Inject
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import org.jetbrains.dokka.DocumentationOptions
+import org.jetbrains.dokka.DokkaLogger
+import org.jetbrains.dokka.DokkaResolutionFacade
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import java.util.*
+
+open class KotlinWebsiteSampleProcessingService
+@Inject constructor(options: DocumentationOptions,
+                    logger: DokkaLogger,
+                    resolutionFacade: DokkaResolutionFacade)
+    : DefaultSampleProcessingService(options, logger, resolutionFacade) {
+
+
+    fun buildReplacementData(psiElement: PsiElement): Map<TextRange, String> {
```

Comment:
Why do you need such a complicated scheme? Why can't you just go directly over the PSI and produce output text in a single pass, replacing assertions with prints as needed?